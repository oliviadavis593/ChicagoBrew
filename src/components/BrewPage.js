import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import STORE from '../components/store';
import Navigation from '../components/Navigation';
import '../styles/BrewPage.css';


class BrewPage extends Component {

    render() {
        //display info from specific brew 
        const { brew_id } = this.props.match.params; 
        const brew = STORE.brews.find(brew => brew.id === parseInt(brew_id, 10));
        return(
         <div>
             <div className='navigation'>
                 <Navigation />
             </div>
             <section className='cbt-brew-page__container'>
             <header>
               <h1>{brew.name}</h1>
             </header>
             <div className='cbt-brew-info__container'>
                 <div className='cbt-brew_image'>
                    
                 </div>
                <div className='cbt-brew_phone-number'>
                    <span>
                        Phone:
                        <br />
                    </span>
                   {brew.phone_number}
                </div>
                <div className='cbt-brew_address'>
                    <span>
                        Address:
                        <br />
                    </span>
                   {brew.address}
                </div>
                <div className='cbt-brew_website'>
                   <a href={brew.website} target='blank'>Website</a>
                </div>
                <details closed='true' className='cbt-brew_details'>
                        <summary>
                            Details
                        </summary>
                        {brew.details}
                </details>
             </div>
             <div className='cbt-add-brew-item'>
                    <button 
                    tag={Link}
                    to='/mybrews' 
                    className='cbt-add-button'
                    onClick={this.props.addBrew}
                    >
                        Add to list
                    </button>
             </div>
         </section>  
         </div> 
        )
    }
}

export default BrewPage; 