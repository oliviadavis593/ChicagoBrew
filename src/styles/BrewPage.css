summary {
    outline: none;
    font-family: 'Metal Mania', cursive; 
    font-size: 20px; 
}

details {
    font-family: 'Roboto', sans-serif;
    font-size: 15px; 
}

.cbt-brew-page__container {
    border: 1px solid #df4811; 
    background-color: #df4811; 
    width: 300px; 
    margin: 0 auto; 
    font-size: 20px; 
}

.cbt-brew-info__container {
    width: 250px; 
    margin: 0 auto; 
}

.cbt-brew_phone-number,
.cbt-brew_address,
.cbt-brew_website,
.cbt-brew_details {
    margin-top: 40px; 
}


.cbt-add-brew-item {
    margin-top: 40px;
    margin-bottom: 40px;  
}

.cbt-brew_website a {
    color: #fff; 
}