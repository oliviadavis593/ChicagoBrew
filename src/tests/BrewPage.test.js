import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import BrewPage from '../components/BrewPage';


const data = {params: {name: 'test', phone_number: 'ddd', address: 'test', details: 'test', website: 'sss'}};

describe('BrewPage component', () => {
    it('render without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><BrewPage match={ {...data}}/></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })
    
    it('renders the UI as expected', () => {
        const tree = renderer 
            .create(<BrowserRouter><BrewPage match={ {...data} }/></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })

})