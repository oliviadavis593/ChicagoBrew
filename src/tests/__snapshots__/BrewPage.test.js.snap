// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrewPage component renders the UI as expected 1`] = `
<div>
  <div
    className="navigation"
  >
    <div
      className="cbt-navigation"
    >
      <div
        className="cbt-logo"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <img
            alt="cb-logo"
            className="cb-logo"
            src="chicagobrew-logo.png"
          />
        </a>
      </div>
      <nav
        className="nav"
      >
        <a
          aria-current={null}
          className="cbt-link__home"
          href="/dashboard"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="cbt-link__mybrew"
          href="/mybrews"
          onClick={[Function]}
        >
          My Brew
        </a>
      </nav>
    </div>
  </div>
  <section
    className="cbt-brew-page__container"
  >
    <header>
      <h1 />
    </header>
    <div
      className="cbt-brew-info__container"
    >
      <div
        className="cbt-brew_image"
      />
      <div
        className="cbt-brew_phone-number"
      >
        <span>
          Phone:
          <br />
        </span>
      </div>
      <div
        className="cbt-brew_address"
      >
        <span>
          Address:
          <br />
        </span>
      </div>
      <div
        className="cbt-brew_website"
      >
        <a
          target="blank"
        >
          Website
        </a>
      </div>
      <details
        className="cbt-brew_details"
        closed="true"
      >
        <summary>
          Details
        </summary>
      </details>
    </div>
    <div
      className="cbt-add-brew-item"
    >
      <button
        className="cbt-add-button"
        onClick={[Function]}
        tag={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        to="/mybrews"
      >
        Add to list
      </button>
    </div>
  </section>
</div>
`;
